situp-plugins/elasticsearch/src/main/java/com/amazon/situp/plugins/sink/elasticsearch/ConnectionConfiguration.java

```diff
@@ -136,22 +115,14 @@ public RestHighLevelClient createClient() {
     final RestClientBuilder restClientBuilder = RestClient.builder(httpHosts);
     final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
     if (username != null) {
+      LOG.info("Using the username provided in the config.");
       credentialsProvider.setCredentials(
               AuthScope.ANY, new UsernamePasswordCredentials(username, password));
     }
-    // TODO: customize trustStrategy?
-    final TrustStrategy trustStrategy = new TrustAllStrategy();
-    final SSLContext sslContext;
-    try {
-      // TODO: load KeyStore
-      sslContext = SSLContexts.custom().loadTrustMaterial(null, trustStrategy).build();
-    } catch (Exception e) {
-      throw new RuntimeException(e.getMessage(), e);
-    }
+    final SSLContext sslContext = certPath != null ? getCAStrategy(certPath) : getTrustAllStrategy();
     restClientBuilder.setHttpClientConfigCallback(
             httpClientBuilder -> httpClientBuilder
                     .setDefaultCredentialsProvider(credentialsProvider)
-                    .setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE)
```

Comment:
Just recalled that one thing I have not figured out is whether the hostname verification will still work for a single CA if there are multiple nodes, node-0, node-1 that have different host urls.